client/my-sites/email/email-providers-comparison/price/discount-or-offer-terms.tsx

```diff
@@ -0,0 +1,95 @@
+import { useTranslate } from 'i18n-calypso';
+import { ProductListItem } from 'calypso/state/products-list/selectors/get-products-list';
+
+import './discount-or-offer-terms.scss';
+
+type DiscountOrOfferTermsProps = {
+	isDiscounted: boolean;
+	isEligibleForIntroductoryOffer: boolean;
+	product: ProductListItem | null;
+	isEligibleForIntroductoryOfferFreeTrial: boolean;
+};
+
+const DiscountOrOfferTerms = ( {
+	isDiscounted,
+	isEligibleForIntroductoryOffer,
+	isEligibleForIntroductoryOfferFreeTrial,
+	product,
+}: DiscountOrOfferTermsProps ) => {
+	const translate = useTranslate();
+
+	// This only makes sense if there's some sale discount or introductory offer
+	if ( ! isDiscounted && ! isEligibleForIntroductoryOffer ) {
+		return null;
+	}
+
+	const introductoryOffer = product?.introductory_offer;
+	const intervalUnit = String( introductoryOffer?.interval_unit );
+	const intervalCount = Number( introductoryOffer?.interval_count );
+	const numberOfDiscountedRenewals = Number( introductoryOffer?.transition_after_renewal_count );
+	const numberOfDiscountedIntervals = intervalCount + numberOfDiscountedRenewals;
+
+	return (
+		<>
+			{ isEligibleForIntroductoryOfferFreeTrial && (
+				<div className="discount-or-offer-terms__free-trial-badge discount-or-offer-terms__item badge badge--info-green">
+					{ intervalUnit === 'month'
+						? translate( '%(numberOfMonths)d month free', '%(numberOfMonths)d months free', {
+								args: {
+									numberOfMonths: numberOfDiscountedIntervals,
+								},
+								count: numberOfDiscountedIntervals,
+						  } )
+						: translate( '%(numberOfYears)d year free', '%(numberOfYears)d years free', {
+								args: {
+									numberOfYears: numberOfDiscountedIntervals,
+								},
+								count: numberOfDiscountedIntervals,
+						  } ) }
+				</div>
+			) }
+
+			{ isEligibleForIntroductoryOffer && ! isEligibleForIntroductoryOfferFreeTrial && (
+				<div className="discount-or-offer-terms__introductory-offer-badge discount-or-offer-terms__item badge badge--info-green">
+					{ intervalUnit === 'month'
+						? translate(
+								'%(numberOfMonths)d month discount',
+								'%(numberOfMonths)d months discount',
+								{
+									args: {
+										numberOfMonths: numberOfDiscountedIntervals,
+									},
+									count: numberOfDiscountedIntervals,
+								}
+						  )
+						: translate( '%(numberOfYears)d year discount', '%(numberOfYears)d years discount', {
+								args: {
+									numberOfYears: numberOfDiscountedIntervals,
+								},
+								count: numberOfDiscountedIntervals,
+						  } ) }
+				</div>
+			) }
+
+			{ isDiscounted && (
+				<div className="discount-or-offer-terms__discount-badge discount-or-offer-terms__item discount-or-offer-terms__discounted_item badge badge--info-green">
+					{ isEligibleForIntroductoryOffer
+						? translate( '+ Extra %(discount)d%% off', {
```

Comment:
I wonder if we can simplify this by passing `+ Extra` and `Limited time:` as an `arg`

```
translate( '%(label)s%% %(discount)d%% off', {
	args: {
		label: isEligibleForIntroductoryOffer ? 'Extra' : 'Limited time:',
		discount: product?.sale_coupon?.discount,
	},
	comment: "%(label)s describes the offer (it can be 'Extra' when multiple offers are in place or 'Limited time' for a limited time offer) %(discount)d is a numeric discount percentage (e.g. '40')",
} )
```

Or does this complicate things for translation? 

In terms of the actual label itself, I see the reasoning behind saying `+ Extra`, since we are *adding* an extra 20% off of the introductory offer. However, I feel throwing in the plus operator `+` could create confusing at first glance, since typically discounts operates at `-` (minus an amount). To avoid confusion, perhaps having `Extra` (without the `+`) would do.

Reply:
> Or does this complicate things for translation?

IMO yes, but I can ping @dlind1 for a more informed opinion.

- For one, `placeholders` should be items that are dynamic, like the cost of a product, or a potential domain name purchase.
- The proposed `placeholders` themselves are not of the same kind. One is an adjective which describes another `placeholders`, while the other contains a colon.

However, since there are just two possibilities then it's just simpler to have two separate instances.

Reply:
> To avoid confusion, perhaps having Extra (without the +) would do.

OK

Reply:
In this case `label: isEligibleForIntroductoryOffer ? 'Extra' : 'Limited time:'` would mean that `Extra` and `Limited time` won't be translated. It's best to not concatenate strings or pass translatable strings as arguments, so having two strings would be the best solution, e.g.
`translate( 'Extra %(discount)d%% off')` and `translate( 'Limited time: %(discount)d%% off')`